### PR TITLE
fix: dead links; add ci link check for docs

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

/kind cleanup
/kind documentation

**What this PR does / why we need it**:

fix some dead docs links, also add a CI check for dead docs. maybe we should run it from prow for pre-merge approvals? but github action felt simple enough. LMK what you think 😄 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [x] includes documentation (heh)
- [x] adds doc** tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/book/src/topics/troubleshooting.md [hunk 1]
```diff
@@ -155,10 +155,10 @@ ssh -J capi@${apiserver} capi@${node}
...
 
-As part of [CI](../../../../scripts/ci-e2e.sh) there is a [log collection script](hack/../../hack/log/log-dump.sh) which you can also leverage to pull all the logs for machines which will dump logs to `${PWD}/_artifacts}` by default:
+As part of [CI](../../../../scripts/ci-e2e.sh) there is a [log collection script](../../../../hack/log/log-dump.sh) which you can also leverage to pull all the logs for machines which will dump logs to `${PWD}/_artifacts}` by default:
 
 ```bash
 ./hack/log/log-dump.sh
 ```
 
-There are also some [provided scripts](/hack/debugging/Readme.md) that can help automate a few common tasks.
+There are also some [provided scripts](../../../../hack/debugging/Readme.md) that can help automate a few common tasks.
```
Comment (alexeldeib) [inline docs/book/src/topics/troubleshooting.md:164]: this is valid but the crawler had a card time with absolute paths.

**Comment by alexeldeib** [comment]
I just opened a similar PR against krustlet, they have an additional markdown linter enabled but it doesn't seem to do link checking. maybe we want that too? 

https://github.com/alexeldeib/krustlet/blob/a3a384d868490be9c14344aa340416d8504fcb47/.github/workflows/lint.yml#L22-L23

File: .markdownlinkcheck.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "ignorePatterns": [{
+        "pattern": "^https://calendar.google.com/calendar"
```
Comment (alexeldeib) [inline .markdownlinkcheck.json:3]: not friendly to being crawled with this tool, maybe some useragent or header issue

File: .github/workflows/lint-docs.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  lint_docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+      - name: Install markdown-link-check
+        run: npm install -g markdown-link-check
+      - name: markdown-link-check
```
Comment (alexeldeib) [inline .github/workflows/lint-docs.yaml:20]: can stick this in a makefile so prow can run it? thoughts?

**Comment by mboersma** [comment]
💯 to this--links are always going stale and manually verifying is tedious.

I think having all the CI tests for pull requests in `prow` is more consistent. Although I like GH actions a lot, and you've already got it working...

**Comment by CecileRobertMichon** [comment]
+1 for Github Actions

I think we should also move some of the other stuff that doesn't required creds like lint to GH actions (like https://github.com/kubernetes-sigs/cluster-api/pull/4374)

File: docs/book/src/topics/failure-domains.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Failure Domains
 
-The Azure provider includes the support for [failure domains](https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha4.html#optional-support-failure-domains) introduced as part of v1alpha4.
+The Azure provider includes the support for [failure domains](https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html#optional-support-failure-domains) introduced as part of v1alpha4.
```
Comment (CecileRobertMichon) [inline docs/book/src/topics/failure-domains.md:3]:
```suggestion
The Azure provider includes the support for [failure domains](https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html#optional-support-failure-domains) introduced as part of v1alpha3.
```

while you're here

**Review by CecileRobertMichon — APPROVED** [review]
lgtm

**Comment by alexeldeib** [comment]
> I think having all the CI tests for pull requests in prow is more consistent.

Yeah, my main concern is whether this ends up messing with presubmit/merge approvals. If it does, it's no sweat to switch it to prow. We do already have GH actions for post-merge coverage, I didn't think it's a huge issue to mix.

**Comment by CecileRobertMichon** [comment]
> Yeah, my main concern is whether this ends up messing with presubmit/merge approvals. If it does, it's no sweat to switch it to prow. We do already have GH actions for post-merge coverage, I didn't think it's a huge issue to mix.

it should be fine, we just need to make the job required in GH settings (which I can do as an admin of the repo). Will leave it optional until it's merged and has run successfully on other PRs though.

**Comment by alexeldeib** [comment]
Repeated 404s installing the azure cli on buster?

**Comment by CecileRobertMichon** [comment]
> Repeated 404s installing the azure cli on buster?

weird, haven't seen this anywhere else

**Comment by alexeldeib** [comment]
/test pull-cluster-api-provider-azure-e2e-windows


<br class="Apple-interchange-newline">

**Comment by alexeldeib** [comment]
Some kind of dns issue maybe? I’m able to hit the 404’d url and that version of that package looks to be in the repo…

**Comment by alexeldeib** [comment]
hm, I actually am getting different results in different clients...works on mobile but 404 on desktop :/

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Comment by nader-ziada** [comment]
looks like a flake in e2e

/retest

**Comment by CecileRobertMichon** [comment]
looks like packages.microsoft.com is still having issues

**Comment by CecileRobertMichon** [comment]
/retest

timeout due to packages.microsoft.com added latency

**Pull request merged**